### PR TITLE
Fix: Advanced Account Rule for Fees under Accrual Periodic type (Fineract-1031)

### PR DESCRIPTION
**Pull request by BLasan**
## Description

See [FINERACT - 1031](https://issues.apache.org/jira/browse/FINERACT-1031)

## Checklist

Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Write the commit message as per https://github.com/apache/fineract/#pull-requests

- [x] Acknowledge that we will not review PRs that are not passing the build _("green")_ - it is your responsibility to get a proposed PR to pass the build, not primarily the project's maintainers.

- [ ] Create/update unit or integration tests for verifying the changes made.

- [x] Follow coding conventions at https://cwiki.apache.org/confluence/display/FINERACT/Coding+Conventions.

- [ ] Add required Swagger annotation and update API documentation at fineract-provider/src/main/resources/static/api-docs/apiLive.htm with details of any API changes

- [x] Submission is not a "code dump".  (Large changes can be made "in repository" via a branch.  Ask on the developer mailing list for guidance, if required.)

FYI our guidelines for code reviews are at https://cwiki.apache.org/confluence/display/FINERACT/Code+Review+Guide.

### CODE REVIEW

**Comment by BLasan** [comment]
@goutham-M @thesmallstar  Would you mind reviewing this?

File: fineract-provider/src/main/java/org/apache/fineract/accounting/journalentry/service/AccrualBasedAccountingProcessorForLoan.java [hunk 1]
```diff
@@ -215,17 +215,21 @@ private void createJournalEntriesForRepaymentsAndWriteOffs(final LoanDTO loanDTO
...
+                        AccrualAccountsForLoan.INCOME_FROM_FEES.getValue(), loanProductId, loanId, transactionId, transactionDate,
+                        feesAmount, isReversal, loanTransactionDTO.getFeePayments());
+                // GLAccount account = this.helper.getLinkedGLAccountForLoanProduct(loanProductId,
+                // AccrualAccountsForLoan.INCOME_FROM_FEES.getValue(), paymentTypeId);
+                // if (accountMap.containsKey(account)) {
+                // BigDecimal amount = accountMap.get(account).add(feesAmount);
+                // accountMap.put(account, amount);
+                // } else {
+                // accountMap.put(account, feesAmount);
+                // }
```
Comment (thesmallstar) [inline fineract-provider/src/main/java/org/apache/fineract/accounting/journalentry/service/AccrualBasedAccountingProcessorForLoan.java]: why is the commented code still there?

**Comment by thesmallstar** [comment]
Did you test this locally? Is the issue mentioned solve,
Don't have fineract setup on the new laptop(will do this soon).
@bharathc27 can you please give a green flag here?

File: fineract-provider/src/main/java/org/apache/fineract/accounting/journalentry/service/AccrualBasedAccountingProcessorForLoan.java (hunk 1 above)
Comment (BLasan) [reply fineract-provider/src/main/java/org/apache/fineract/accounting/journalentry/service/AccrualBasedAccountingProcessorForLoan.java]: My bad. Will remove these

**Comment by BLasan** [comment]
> Did you test this locally? Is the issue mentioned solve,
> Don't have fineract setup on the new laptop(will do this soon).
> @bharathc27 can you please give a green flag here?

Yeah, tested locally. The issue was fixed with the changes. @bharathc27 will also test this

**Comment by BLasan** [comment]
@thesmallstar  Would you mind reviewing this?

**Comment by thesmallstar** [comment]
Merging this, still waiting for green flag from bharat.

**Pull request merged**